@here/harp-materials/lib/SolidLineMaterial.ts

```diff
@@ -124,13 +131,40 @@ void main() {
     float segmentDist = mod(vExtrusionCoord.x, dashSize + gapSize) / dashSize;
     float dashDist = 0.5 - distance(segmentDist, halfSegment);
     float dashWidth = fwidth(dashDist);
-    alpha *= smoothstep(-dashWidth, dashWidth, dashDist);
+    float dashedBlendFactor = smoothstep(-dashWidth, dashWidth, dashDist);
+
+    #ifdef USE_DASH_ALPHA
+    if (outlineWidth <= 0.0) {
+      alpha *= dashedBlendFactor;
```

Comment:
You should avoid `if`s in shader code. In this case you can use `step`.
```glsl
alpha *= step(outlineWidth, 0.) + step(-outlineWidth, 0.) * dashedBlendFactor;
```

Reply:
@okanishchev What's the point? Doesn't it make it complicated to read with no performance impact? I see no reason to do that.

Reply:
The branches in shader code can be very slow on some hardware, for example a bit of explanation can be found here: https://stackoverflow.com/questions/37827216/does-if-statements-slow-down-my-shader
And the constructions like `a * (1 - c) + b * c` is pretty usual in shader code, but for readability you can use additional variable:
```glsl
float enableDashedBlend = step(-outlineWidth, 0.);
alpha *= (1. - enableDashedBlend) + enableDashedBlend * dashedBlendFactor;
```

Reply:
@okanishchev Then I guess you should rise a separate topic as if-statement is used in pretty-much any fragmentSource in the project. (see `IconMaterial.ts`, `TextMaterials.ts`, and a lot more)

**UPD:** I will replace it with define (USE_OUTLINE)